aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/UserCredentials.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.annotation.RawRes;
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.testutils.Resources;
+import com.amplifyframework.util.Immutable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+final class UserCredentials implements Iterable<UserCredentials.Credential> {
```

Comment:
this is clean, and likely reusable for other categories like API :)

As far as I know, we lack tests for testing schemas that contain `@auth`

Reply:
To this point -- I'll move this over into `testutils`, instead of `aws-storage-s3`. That way it _can_ be re-used by other modules' tests.